gobblin-runtime/src/main/java/gobblin/runtime/JobState.java

```diff
@@ -564,8 +564,12 @@ public DatasetState newDatasetState(boolean fullCopy) {
   }
 
   /**
-   * A subclass of {@link JobState} that is used to represent dataset states. This class is currently
-   * identical to {@link JobState} except that the name is more meaningful and less confusing.
+   * A subclass of {@link JobState} that is used to represent dataset states.
+   *
+   * <p>
+   *   A {@code DatasetState} does <em>not</em> contain any properties. Operations such as {@link #getProp(String)}
+   *   and {@link #setProp(String, Object)} are not supported.
+   * </p>
    */
   public static class DatasetState extends JobState {
```

Comment:
Because dataset states need to be persisted into the state store, which, by definition, only takes `State` objects. Also, extending `State` and `JobState` enables it to inherit methods such as `read`,  `write`, `addTaskState`, etc.